### PR TITLE
Fixed documentation issues in `access` `dataone` and `editing`

### PR DESCRIPTION
**Pull request by jeanetteclark**
See issue #66

### CODE REVIEW

File: R/dataone.R [hunk 1]
```diff
@@ -51,10 +61,15 @@ get_token <- function(node) {
 
 #' Determine whether the set token is expired.
 #'
-#' @return
+#' @return (boolean)
```
Comment (amoeba) [inline R/dataone.R]: boolean is okay but, in R, booleans are of type `logical`, so I like to stick to that. Would you consider changing this to say `(logical)`?

**Comment by amoeba** [comment]
Excellent, thanks @jeanetteclark! I took a quick look and it all looks good. I did make one minor comment. Would you mind making that change, re-pushing the commit back to your fork and I'll merge it?

**Comment by jeanetteclark** [comment]
@amoeba I wrote boolean because it was in the style of other variable definitions in the package (there are a lot of boolean's in there). Should I change them all?

**Comment by amoeba** [comment]
Ah, of course I wrote them that way. No, we can keep this PR straight to the point.

**Comment by amoeba** [comment]
Thanks for the changes!

I saw you removed the @export from `sysmeta_to_other_entity`. You probably saw the `Deprecated` call inside it. I did that because I renamed the function but it's not nice for your users to just rename functions between package versions. So I left the old function name as is but replaced its body with a `Deprecated` call. Given that it's been a while since I deprecated it, I bet no one would even call it anymore so I'm just going to remove it entirely. Not a big deal either way.

**Pull request merged**